Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnCancel.hs

```diff
@@ -165,7 +169,7 @@ tfFulfillments booking driverQuote customerPhoneNo rideStatus mbVehicle = do
           { fulfillmentStateDescriptor =
               Just $
                 Spec.Descriptor
-                  { descriptorCode = Just . show $ BUtils.mapRideStatus rideStatus',
+                  { descriptorCode = fmap (show . BUtils.mapRideStatus) rideStatus',
```

Comment:
This is a mandatory field in ONDC spec.
If there is no driver assigned as of now, as part of v2.0.0 spec we should send `RIDE_ASSIGNED` code and the same will be sent when driver is assigned as well

Reply:
you can update `mapRideStatus` to take `Maybe RideStatus`